### PR TITLE
pd-ctl, server, api: support namespace config

### PR DESCRIPTION
**Pull request by Connor1996**

### CODE REVIEW

File: pdctl/command/config_command.go [hunk 1]
```diff
@@ -24,9 +24,10 @@ import (
 )
 
 var (
-	configPrefix    = "pd/api/v1/config"
-	schedulePrefix  = "pd/api/v1/config/schedule"
-	replicatePrefix = "pd/api/v1/config/replicate"
+	configPrefix      = "pd/api/v1/config"
+	schedulePrefix    = "pd/api/v1/config/schedule"
+	replicationPrefix = "pd/api/v1/config/replicate"
+	namespacePrefix   = "pd/api/v1/config/namespace/%s"
```
Comment (siddontang) [inline pdctl/command/config_command.go]: remove %s

File: pdctl/command/config_command.go [hunk 2]
```diff
@@ -61,13 +74,34 @@ func NewShowAllConfigCommand() *cobra.Command {
...
 	}
+	sc.AddCommand(NewSetNamespaceConfigCommand())
+	return sc
+}
+
+// NewSetNamespaceConfigCommand a set subcommand of set subcommand
+func NewSetNamespaceConfigCommand() *cobra.Command {
+	sc := &cobra.Command{
+		Use:   "namespace <name> <option> <value>",
+		Short: "show namespace config of PD",
```
Comment (disksing) [inline pdctl/command/config_command.go]: Wrong desc.

File: server/cluster.go [hunk 3]
```diff
@@ -191,6 +196,45 @@ func (s *Server) SetReplicationConfig(cfg ReplicationConfig) {
...
+	} else {
+		s.scheduleOpt.ns[name] = newNamespaceOption(&cfg)
+		s.scheduleOpt.persist(s.kv)
+		log.Infof("namespace:%v config is added: %+v", name, cfg)
+	}
+
+	s.cfg.Namespace[name] = cfg
+}
+
+// IsNamespaceExist returns
```
Comment (disksing) [inline server/cluster.go]: incomplete comment.

File: server/config.go [hunk 4]
```diff
@@ -22,6 +22,8 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/pingcap/pd/server/namespace"
+
```
Comment (disksing) [inline server/config.go]: redundant empty line.

File: server/config.go [hunk 5]
```diff
@@ -398,15 +403,107 @@ func (c *ReplicationConfig) adjust() {
 	adjustUint64(&c.MaxReplicas, defaultMaxReplicas)
 }
 
+// NamespaceConfig is to cover the global setting for specific namespace
```
Comment (disksing) [inline server/config.go]: cover -> overwrite?

File: server/api/router.go [hunk 6]
```diff
@@ -55,6 +55,8 @@ func createRouter(prefix string, svr *server.Server) *mux.Router {
 	router.HandleFunc("/api/v1/config/schedule", confHandler.GetSchedule).Methods("GET")
 	router.HandleFunc("/api/v1/config/replicate", confHandler.SetReplication).Methods("POST")
 	router.HandleFunc("/api/v1/config/replicate", confHandler.GetReplication).Methods("GET")
+	router.HandleFunc("/api/v1/config/namespace/{name}", confHandler.GetNamespace).Methods("GET")
+	router.HandleFunc("/api/v1/config/namespace/{name}", confHandler.SetNamespace).Methods("POST")
```
Comment (disksing) [inline server/api/router.go]: Do we need a delete API?

Comment (disksing) [reply server/api/router.go]: Need some test for it?

File: pdctl/command/config_command.go (hunk 1 above)
Comment (Connor1996) [reply pdctl/command/config_command.go]: it is used to indicate name of namespace

Comment (disksing) [reply pdctl/command/config_command.go]: You can avoid '%s' by using string concat or _path.Join_.

File: server/schedule/scheduler.go [hunk 7]
```diff
@@ -45,6 +45,7 @@ type Cluster interface {
 	RegionReadStats() []*core.RegionStat
 
 	// get config methods
+	GetOpt() NamespaceOptions
```
Comment (nolouch) [inline server/schedule/scheduler.go]: `GetNamespaceOpt`?

File: server/api/router.go (hunk 6 above)
Comment (Connor1996) [reply server/api/router.go]: add it and test it manually

File: server/schedule/scheduler.go (hunk 7 above)
Comment (Connor1996) [reply server/schedule/scheduler.go]: cause here means return the opt field of itself, I think GetOpt() is clear

**Comment by disksing** [comment]
The rest LGTM.

**Comment by disksing** [comment]
Fix #841.

**Pull request merged**